Source/NSTextFinder.m

```diff
@@ -0,0 +1,333 @@
+/* Implementation of class NSTextFinder
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 02-08-2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSArchiver.h>
+
+#import "AppKit/NSTextFinder.h"
+#import "AppKit/NSTextView.h"
+#import "AppKit/NSUserInterfaceValidation.h"
+
+#import "GSTextFinder.h"
+
+@implementation NSTextFinder
+
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      // initialize finder...
+      _finder = [[GSTextFinder alloc] init];
```

Comment:
In the long run I would expect that this class should replace GSTextFinder, but for now this reuse might work. Not really sure.

Reply:
That was my read on this change as well.  It also seems that NSTextCheckingController is meant as a replacement for NSSpellChecker's role as a controller to the spell checking panel which was always awkward in the first place.

Reply:
I'll do it until it no longer makes sense, I want to introduce as little risk as possible since this is a rather sensitive portion of the system.